.github/workflows/python-docs.yml

```diff
@@ -0,0 +1,27 @@
+name: documentation (python)
+
+on: [push, pull_request, workflow_dispatch]
```

Comment:
normally we don't run on push AND pull_request, since that results in the workflow running twice for each commit. so the `push` action is usually limited to `main` branch.